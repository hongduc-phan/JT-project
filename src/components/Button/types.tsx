export enum ButtonVariants {
  Primary = 'primary',
  Secondary = 'secondary',
  Icon = 'icon',
}
