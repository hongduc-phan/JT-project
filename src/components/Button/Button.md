Button example:

```js
const Button = require('./').default;
<div className="button-row">
  <Button>Borderless</Button>
  <Button variant="secondary">Secondary</Button>
  <Button variant="primary">Primary</Button>
</div>;
```

Button disabled

```js
const Button = require('./').default;
<div className="button-row">
  <Button disabled={true}>Borderless</Button>
  <Button disabled={true} variant="secondary">
    Secondary
  </Button>
  <Button disabled={true} variant="primary">
    Primary
  </Button>
</div>;
```

Button display size:

```js
const Button = require('./').default;
<div className="button-row">
  <Button variant="secondary" display="big">
    Secondary Big Button
  </Button>
  <Button variant="primary" display="big">
    Primary Big Button
  </Button>
</div>;
```

Button with HREF

```js
const Button = require('./').default;
<div className="button-row">
  <Button
    variant="primary"
    display="big"
    href="https://juztalent.com"
    role="button"
    target="__blank"
    rel="noopener noreferrer"
  >
    Primary Big Button
  </Button>
</div>;
```
