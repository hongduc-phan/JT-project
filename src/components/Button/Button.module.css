.wrap {
  color: var(--blue-default);
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  padding-top: 0.6875rem;
  padding-bottom: 0.6875rem;
  background: transparent;
  border: 0;
  border-radius: 0.25rem;
  outline: transparent;
  min-width: 6rem;
  font-weight: var(--font-weight-bold);
  cursor: pointer;
  transition: 150ms ease-in-out;
  text-decoration: none;
  display: inline-block;
}

.wrap:hover,
.wrap:active,
.wrap:hover {
  background-color: var(--grey-offwhite);
}

.wrap:disabled,
.wrap:disabled:hover {
  color: var(--grey-light);
  background-color: transparent;
}

.secondary {
  border: 1px solid var(--grey-medium);
  color: var(--grey-medium);
}

.icon {
  display: inline-flex;
  min-width: 0;
  padding: 0.5rem;
  margin: 0 0.25rem;
  border-radius: 50%;
  font-size: 1.5rem;
  color: #9ca2a7;
}

.icon:hover,
.icon:active {
  color: var(--white);
  background-color: #9ca2a7;
}

.icon:disabled {
  opacity: 0.5;
}

.secondary:hover,
.secondary:active {
  border: 1px solid var(--grey-medium);
  background-color: var(--grey-offwhite);
}

.secondary:disabled {
  opacity: 0.5;
  background-color: transparent;
}

.primary {
  border: 1px solid var(--blue-default);
  color: var(--white);
  background-color: var(--blue-default);
}

.primary:hover,
.primary:active {
  color: var(--white);
  background-color: var(--blue-default);
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.16);
}

.primary:disabled {
  background-color: var(--grey-offwhite);
  border: 1px solid var(--grey-offwhite);
  color: var(--grey-light);
  box-shadow: none;
}

.big {
  padding-top: 1.0625rem;
  padding-bottom: 1.0625rem;
}
