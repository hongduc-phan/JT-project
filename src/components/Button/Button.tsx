import React, {AllHTMLAttributes, ReactNode} from 'react';
import clsx from 'clsx';

import Typo from '../Typo';
import styles from './Button.module.css';
import {ButtonVariants} from './types';

import {TypoVariants, TypoColors, TypoAlignment} from '../Typo';

export interface IButtonProps
  extends AllHTMLAttributes<HTMLButtonElement | HTMLAnchorElement> {
  children: ReactNode;
  /** Class name append to root button */
  className?: string;
  variant?: ButtonVariants;
  display?: 'big';
  href?: string;
}

const Button = ({
  children,
  className,
  variant,
  display,
  href,
  ...other
}: IButtonProps) => (
  <Typo<HTMLButtonElement | HTMLAnchorElement>
    {...other}
    tag={undefined === href ? 'button' : 'a'}
    color={TypoColors.inherit}
    variant={TypoVariants.button}
    align={TypoAlignment.middle}
    href={href}
    className={clsx(
      styles.wrap,
      className,
      variant === ButtonVariants.Icon && styles.icon,
      variant === ButtonVariants.Secondary && styles.secondary,
      variant === ButtonVariants.Primary && styles.primary,
      display === 'big' && styles.big,
    )}
  >
    {children}
  </Typo>
);

export default Button;
