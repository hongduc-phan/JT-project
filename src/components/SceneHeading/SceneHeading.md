Example

```js
const SceneHeading = require('./').default;
const Button = require('../Button').default;

<div>
  <SceneHeading
    title="Tile here"
    subTitle="Subtitle here"
    renderActions={<Button variant="primary">Action</Button>}
  />
</div>;
```
