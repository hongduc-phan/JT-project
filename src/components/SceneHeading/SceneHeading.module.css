.wrapper {
  padding: 1.5rem;
}

.heading {
  margin-bottom: 1.5rem;
}

.title {
  margin-bottom: 0.75rem;
}

.colRight {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-end;
  height: 100%;
}

.row {
  display: flex;
  flex-wrap: wrap;
}

.col1 {
  flex: 0 0 60%;
  max-width: 60%;
}

.col2 {
  flex: 0 0 40%;
  max-width: 40%;
}
