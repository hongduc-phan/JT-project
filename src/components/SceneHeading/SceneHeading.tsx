import React, {FunctionComponent, ReactNode} from 'react';
import clsx from 'clsx';
import Typo, {TypoColors, TypoVariants} from '../Typo';
import styles from './SceneHeading.module.css';

export interface SceneHeadingProps {
  title: string;
  subTitle: string;
  className?: string;
  renderActions?: ReactNode;
}

const SceneHeading: FunctionComponent<SceneHeadingProps> = ({
  title,
  subTitle,
  renderActions,
  className,
}: SceneHeadingProps) => {
  return (
    <div className={clsx(styles.heading, className)}>
      <div className={styles.row}>
        <div className={styles.col1}>
          <Typo tag="div" variant={TypoVariants.h2} className={styles.title}>
            {title}
          </Typo>
          <Typo
            tag="div"
            color={TypoColors.greyDark}
            className={styles.subTitle}
          >
            {subTitle}
          </Typo>
        </div>
        <div className={styles.col2}>
          <div className={styles.colRight}>{renderActions}</div>
        </div>
      </div>
    </div>
  );
};

export default SceneHeading;
