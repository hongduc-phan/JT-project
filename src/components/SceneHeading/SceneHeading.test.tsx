import React from 'react';
import renderer from 'react-test-renderer';

import SceneHeading from './SceneHeading';
import Button, {ButtonVariants} from '../Button';

describe('<SceneHeading />', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(
        <SceneHeading
          title="Tile here"
          subTitle="Subtitle here"
          renderActions={
            <Button variant={ButtonVariants.Primary}>Action</Button>
          }
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
