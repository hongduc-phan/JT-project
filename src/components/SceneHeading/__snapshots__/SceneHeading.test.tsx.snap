// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SceneHeading /> Should match snapshot 1`] = `
<div
  className="heading"
>
  <div
    className="row"
  >
    <div
      className="col1"
    >
      <div
        className="typo title h2 black left"
      >
        Tile here
      </div>
      <div
        className="typo subTitle body1 greyDark left"
      >
        Subtitle here
      </div>
    </div>
    <div
      className="col2"
    >
      <div
        className="colRight"
      >
        <button
          className="typo wrap primary button inherit middle"
        >
          Action
        </button>
      </div>
    </div>
  </div>
</div>
`;
